<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HashMap on Stack</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Function:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="_functioncomboBox">
          <item>
           <property name="text">
            <string>Insert</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Operator []</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Erase</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Contains</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Generate Keys: -50 to 50</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Extract Negative Keys</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QStackedWidget" name="_paramsStackedWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="Line" name="line_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="formLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Key:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="_insertKeySpinBox">
                <property name="maximum">
                 <number>999999999</number>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Value:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="_insertValueLineEdit"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="_insertPushButton">
                <property name="text">
                 <string>Insert</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Line" name="line_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>Result:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="_insertResultLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>2</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>372</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="Line" name="line_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="formLayout_5">
              <item row="0" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Key:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="_getKeySpinBox">
                <property name="maximum">
                 <number>999999999</number>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Value:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="_getValueLineEdit"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="_getPushButton">
                <property name="text">
                 <string>[]</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Line" name="line_6">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>Result:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="_getResultLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>2</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>372</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_5">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QFormLayout" name="formLayout_6">
              <item row="0" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Key:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="_eraseKeySpinBox">
                <property name="maximum">
                 <number>999999999</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <spacer name="horizontalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="_erasePushButton">
                <property name="text">
                 <string>Erase</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Result:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="_eraseResultLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>2</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_4">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>410</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_6">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QFormLayout" name="formLayout_7">
              <item row="0" column="0">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Key:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="_containsKeySpinBox">
                <property name="maximum">
                 <number>999999999</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <spacer name="horizontalSpacer_8">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="_containsPushButton">
                <property name="text">
                 <string>Contains</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Line" name="line_5">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLabel" name="label_14">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Result:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="_containsResultLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>2</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_5">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>410</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_7">
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QPushButton" name="_generateRandomPushButton">
              <property name="text">
               <string>Generate</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="line_9">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QLabel" name="label_15">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Result:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="_generateResultLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>2</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_6">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>444</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_8">
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QPushButton" name="_extractNegativePushButton">
              <property name="text">
               <string>Extract</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="line_8">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="_extractResultGroupBox">
              <property name="title">
               <string>Result:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <property name="leftMargin">
                <number>4</number>
               </property>
               <property name="topMargin">
                <number>4</number>
               </property>
               <property name="rightMargin">
                <number>4</number>
               </property>
               <property name="bottomMargin">
                <number>4</number>
               </property>
               <item row="0" column="0">
                <widget class="QTreeWidget" name="_extractResultTreeWidget">
                 <column>
                  <property name="text">
                   <string>Key</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>Value</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTreeWidget" name="_hashTreeWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>4</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <column>
        <property name="text">
         <string>Key</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Value</string>
        </property>
       </column>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="_statusBar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>_functioncomboBox</sender>
   <signal>activated(int)</signal>
   <receiver>_paramsStackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_insertPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>insertPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_getPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>getPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>-22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_erasePushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>erasePushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>-13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_containsPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>containsPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>-4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_generateRandomPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generatePushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>44</x>
     <y>-16</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_extractNegativePushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>extractPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>56</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>insertPushButtonClicked()</slot>
  <slot>getPushButtonClicked()</slot>
  <slot>erasePushButtonClicked()</slot>
  <slot>containsPushButtonClicked()</slot>
  <slot>generatePushButtonClicked()</slot>
  <slot>extractPushButtonClicked()</slot>
 </slots>
</ui>
